static/app/views/replays/detail/ai/useFetchReplaySummary.tsx

```diff
@@ -21,23 +21,45 @@ export interface SummaryResponse {
 function createAISummaryQueryKey(
   orgSlug: string,
   projectSlug: string | undefined,
-  replayId: string
+  replayId: string,
+  query: Record<string, any> = {}
 ): ApiQueryKey {
   return [
     `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/summarize/breadcrumbs/`,
+    {query},
   ];
 }
 
-export function useFetchReplaySummary(options?: UseApiQueryOptions<SummaryResponse>) {
+function useAISummaryQueryKey(query: Record<string, any> = {}): ApiQueryKey {
   const organization = useOrganization();
   const {replay} = useReplayContext();
   const replayRecord = replay?.getReplay();
   const project = useProjectFromId({project_id: replayRecord?.project_id});
-  return useApiQuery<SummaryResponse>(
-    createAISummaryQueryKey(organization.slug, project?.slug, replayRecord?.id ?? ''),
-    {
-      staleTime: 0,
-      ...options,
-    }
+  return createAISummaryQueryKey(
+    organization.slug,
+    project?.slug,
+    replayRecord?.id ?? '',
+    query
   );
 }
+
+export function useFetchReplaySummary(options?: UseApiQueryOptions<SummaryResponse>) {
+  const queryKey = useAISummaryQueryKey();
+  return useApiQuery<SummaryResponse>(queryKey, {
+    staleTime: 0,
+    ...options,
+  });
+}
+
+/**
+ * Use to force a regenerate of the AI summary, skipping the backend cache.
+ */
+export function useFetchReplaySummaryForceRegenerate(
```

Comment:
I'd probably just keep this as a single hook that accepted a `regenerate` option e.g.

`useFetchReplaySummary({ regenerate: true })`